File: pysrc/fractal/world/fractum/placement/loop.py
from numpy import true_divide


class Environment:
    def resize_memory(self, n):
        exp = get_experience(self.memory)
        self.recall = Recall(env.gspec, n, exp)

    def simple_memory_stats(self):
        mem = get_experience(env)
        nsamples = length(mem)
        ndistinct = length(merge_by_state(mem))
        return nsamples, ndistinct

File: src/fractal/world/fractum/placement/loop.rs
from numpy import true_divide


pub trait Environment {
    fn resize_memory(&self, n) {
        let exp = get_experience(&self.memory);
        &self.recall = Recall(env.gspec, n, exp);
    }

    fn simple_memory_stats(&self) {
        let mem = get_experience(env);
        
        let nsamples = length(mem);
        let ndistinct = length(merge_by_state(mem));
        
        nsamples, ndistinct
    }
}
